### PR TITLE
zydis 2.0.2 (new formula)

### PR DESCRIPTION
**Pull request by nl5887**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [0] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/zydis.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      system "make", "install"
+      bin.install "ZydisDisasm"
+      bin.install "ZydisInfo"
+      bin.install "ZydisFuzzIn"
+      bin.install "ZydisPerfTest"
+    end
+  end
+
+  test do
+    system "#{bin}/ZydisInfo"
```
Comment (fxcoudert) [inline Formula/zydis.rb]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/zydis.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on "cmake" => :build
+
+  def install
+    mkdir "zydis-build" do
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+      bin.install "ZydisDisasm"
+      bin.install "ZydisInfo"
+      bin.install "ZydisFuzzIn"
+      bin.install "ZydisPerfTest"
```
Comment (fxcoudert) [inline Formula/zydis.rb]: Why are these required? If upstream does not typically install those files as part of `make install`, we probably don't want to ship them.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/zydis.rb (hunk 1 above)
Comment (nl5887) [reply Formula/zydis.rb]: improved the test to compare with expected output

File: Formula/zydis.rb (hunk 2 above)
Comment (nl5887) [reply Formula/zydis.rb]: Removed ZydisFuzzIn and ZydisPerfTest. ZydisDisasm and ZydisInfo are apparently both not installed, but useful.

Comment (fxcoudert) [reply Formula/zydis.rb]: Would you please open an issue upstream, asking why these tools are not installed and whether they are considered standard / should be installed by default?

File: Formula/zydis.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/zydis.rb]: It's now a great test

File: Formula/zydis.rb (hunk 2 above)
Comment (nl5887) [reply Formula/zydis.rb]: @athre0z just fixed the issue, it was not meant to not install them. He will make a new release (2.0.3). I will push the formula as soon as he is finished.

Comment (nl5887) [reply Formula/zydis.rb]: fixed!

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @nl5887 for the pull request!
